ranger/core/progress_tracker.py

```diff
@@ -0,0 +1,33 @@
+# This file is part of ranger, the console file manager.
+# License: GNU GPL version 3, see the file "AUTHORS" for details.
+
+"""
+A Progress Tracker of any Loadable that manages all the progress
+related statistics.
+"""
+
+import time
+
+class ProgressTracker(object):
+    def __init__(self, total_items=0):
+        self.total_items = total_items
+        self.done = 0
+        self.start_time = time.time()
+
+    def set_total_items(self, total_items):
```

Comment:
You may want to consider allowing to set `total_items` explicitly (since it's a "public" attribute) instead of having this boilerplatey setter. Many attributes in the standard library also operate like this.